cm2350/e200z7.py

```diff
@@ -544,8 +436,60 @@ def debug_client_detached(self):
         Handles the condition when a gdb client detaches. For now we will halt
         the emulator when this occurs.
         '''
-        self.queueException(intc_exc.GdbClientDetachEvent())
-        self.resume_exec()
+        # if va is already a breakpoint, don't do it!!
+        if va in self._bpdata:
+            return -1   # raise exception
+
+        brkbytes = self.arch.archGetBreakInstr()
+
+        # check for overlapping breakpoints
+        for tva in range(va, va+len(brkbytes)-1):
+            if tva in self._bpdata:
+                return -2   # raise exception
+
+        for tva in range(va - len(brkbytes)+1, va):
+            if tva in self._bpdata:
+                return -3   # raise exception
+
+        # store existing bytes
+        self._bpdata[va] = self.readMemory(va, len(brkbyes))
```

Comment:
It shouldn't be, this is obviously some sort of copy/paste error. it looks like this may be the result of a bad merge of some sort. I am tracking it down now.

Reply:
Fixed strange copy/paste error where some code from an older version of `cm2350/e200_gdb.py` got pasted over the `debug_client_detached()` function in `cm2350/e200z7.py`